omx/gstomxaudioenc.c

```diff
@@ -644,23 +633,48 @@ gst_omx_audio_enc_set_format (GstAudioEncoder * encoder, GstAudioInfo * info)
   GstOMXAudioEnc *self;
   GstOMXAudioEncClass *klass;
   gboolean needs_disable = FALSE;
-  OMX_PARAM_PORTDEFINITIONTYPE port_def;
-  OMX_AUDIO_PARAM_PCMMODETYPE pcm_param;
   gint i;
+  guint n_samples;
+  OMX_PARAM_PORTDEFINITIONTYPE port_def_in;
+  OMX_PARAM_PORTDEFINITIONTYPE port_def_out;
+  OMX_AUDIO_PARAM_PCMMODETYPE pcm_param;
+  OMX_PORT_PARAM_TYPE param;
   OMX_ERRORTYPE err;
 
   self = GST_OMX_AUDIO_ENC (encoder);
   klass = GST_OMX_AUDIO_ENC_GET_CLASS (encoder);
 
   GST_DEBUG_OBJECT (self, "Setting new caps");
 
+  n_samples =
+      klass->get_num_samples (self, self->enc_out_port,
+      gst_audio_encoder_get_audio_info (GST_AUDIO_ENCODER (self)), NULL);
+
   /* Set audio encoder base class properties */
-  gst_audio_encoder_set_frame_samples_min (encoder,
-      gst_util_uint64_scale_ceil (OMX_MIN_PCMPAYLOAD_MSEC,
-          GST_MSECOND * info->rate, GST_SECOND));
-  gst_audio_encoder_set_frame_samples_max (encoder, 0);
+  gst_audio_encoder_set_frame_samples_min (encoder, n_samples);
+  gst_audio_encoder_set_frame_samples_max (encoder, n_samples);
 
-  gst_omx_port_get_port_definition (self->enc_in_port, &port_def);
+  GST_OMX_INIT_STRUCT (&param);
+
+  param.nPorts = 2;
+  param.nStartPortNumber = 0;
+  err =
+      gst_omx_component_set_parameter (self->enc, OMX_IndexParamAudioInit,
+      &param);
+  if (err != OMX_ErrorNone) {
+    GST_ERROR_OBJECT (self, "Failed to set AudioInit parameters: %s (0x%08x)",
+        gst_omx_error_to_string (err), err);
+    return FALSE;
+  }
+
+  GST_OMX_INIT_STRUCT (&param);
```

Comment:
Do you use `param` for anything?

Reply:
Yes, I use it to set the OMX_IndexParamAudioInit parameters and make sure that the number of ports and nStartPortNumber have the correct values.